### PR TITLE
5677 cross platform alias syntax

### PR DESCRIPTION
**Pull request by timotheemoulin**
- Add a new `aliases` configuration entry in the wp-cli.yml file 
- Allow to use `--alias=foo` instead of `@foo` to use an alias
- Keep the classic arg notation working
- Works properly with the local and global configs merged together even if both styles are used

## New wp-cli.yml config entry `aliases`

Example of a working configuration.

Global config file
```yml
@test-1:
  ssh: user@domain-test.com:~/web
```

Local config file
```yml
@test-2:
  ssh: user@domain-test-2.com
  path: ~/web

aliases:
  test-3:
    ssh: user@domain-test-3.com
  @test-4:
    ssh: user@domain-test-4.com
```

When using the `aliases` config entry, there is no need to prefix the alias name with `@`. Every key in this hash is recognized as a valid alias name.

`@all` and other group features seem to be working properly (though I never had them working on Windows due sur sub-PHP process command not knowing where to find the `ssh` command).

## New CLI parameter `--alias`

Instead of using the `@alias` positional parameter, you can now use the `--alias=foo` param.
This improves the Windows CLI compatibility as `@` is a reserved char (like `$` or <code>`</code>) and need to be quoted to work properly.

`--alias=` param works well with both classic alias definition and the new. And it works exactly the same for groups.

### CODE REVIEW

File: php/WP_CLI/Runner.php [hunk 1]
```diff
@@ -73,7 +73,7 @@ class Runner {
 	private $required_files;
```
Comment (danielbachhuber) [inline php/WP_CLI/Runner.php:73]: Can we revert all of the formatting changes? It's hard to review the changes relevant to this PR...

**Review by danielbachhuber — COMMENTED** [review]
Thanks @timotheemoulin !

It'd be great to have a test case or two for this enhancement.

File: php/WP_CLI/Runner.php (hunk 1 above)
Comment (timotheemoulin) [reply php/WP_CLI/Runner.php:73]: Sure.. sorry about that

**Comment by timotheemoulin** [comment]
> It'd be great to have a test case or two for this enhancement.

I'm having issues when running `phpunit tests`, but they seem unrelated.
```sh
PS C:\laragon\_tools\wp-cli> vendor/bin/phpunit --version          
PHPUnit 9.5.24 #StandWithUkraine

PS C:\laragon\_tools\wp-cli> vendor/bin/phpunit tests    

Fatal error: Uncaught Error: Class 'WP_CLI\Tests\TestCase' not found in C:\laragon\_tools\wp-cli\tests\WP_CLI\WpOrgApiTest.php:8
Stack trace:
#0 C:\laragon\_tools\wp-cli\vendor\phpunit\phpunit\src\Util\FileLoader.php(65): include_once()
#1 C:\laragon\_tools\wp-cli\vendor\phpunit\phpunit\src\Util\FileLoader.php(49): PHPUnit\Util\FileLoader::load('C:\\laragon\\_too...')
#2 C:\laragon\_tools\wp-cli\vendor\phpunit\phpunit\src\Framework\TestSuite.php(394): PHPUnit\Util\FileLoader::checkAndLoad('C:\\laragon\\_too...')
#3 C:\laragon\_tools\wp-cli\vendor\phpunit\phpunit\src\Framework\TestSuite.php(524): PHPUnit\Framework\TestSuite->addTestFile('C:\\laragon\\_too...')
#4 C:\laragon\_tools\wp-cli\vendor\phpunit\phpunit\src\Runner\BaseTestRunner.php(98): PHPUnit\Framework\TestSuite->addTestFiles(Array)
#5 C:\laragon\_tools\wp-cli\vendor\phpunit\phpunit\src\TextUI\Command.php(120): PHPUnit\Runner\BaseTestRunner->getTest('C:\\laragon\\_too...', Array)
#6 C:\laragon\_tools\wp-cli\vendor\phpunit\phpunit\src\TextUI\Command.php(96): PHPUn in C:\laragon\_tools\wp-cli\vendor\phpunit\phpunit\src\TextUI\Command.php on line 98
```
Am I missing something?

**Comment by schlessera** [comment]
@timotheemoulin Running the unit tests through `composer phpunit` should work.

However, a lot of the functional tests related to aliases are now failing. You can run these via `composer behat` (and `composer test` will run all the tests, including linting).

Running the Behat tests will require you to configure a database to run the tests within. You can find more information about this here: https://github.com/wp-cli/wp-cli-tests/#the-database-credentials

In case you have trouble getting the tests to run on your local machine, you can already sneak at the failures within GHA here: https://github.com/wp-cli/wp-cli/actions/runs/3139319664/jobs/5124517193

**Comment by danielbachhuber** [comment]
Proceeding with https://github.com/wp-cli/wp-cli/issues/5594 for this repository. I've captured this PR to https://gist.github.com/danielbachhuber/49f8e24819d647e7c2ff55a3c054e27f in case this PR is auto-closed or broken in some way.

**Pull request closed without merging**